app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def new
+    post_id = params[:id]
+    comment = Comment.new
+    respond_to do |format|
+      format.html { render :new_comment, locals: { comment:, post_id: } }
+    end
+  end
+
+  def create
+    post = Post.find(params[:id])
+    comment = Comment.new(text: params[:comment][:text], post:, user: current_user)
+    if comment.save
+      flash[:success] = 'Comment saved successfully'
+      redirect_to "/users/:user_id/posts/#{post[:id]}"
+    else
+      flash.now[:error] = 'Error: Comment could not be saved'
+      render inline: '<p>Error</p>'
+    end
+  end
```

Comment:
- [ ] [optional] It would be nice having a params method that should be private to ensure you can reuse it eg for your create and update method which uses the same parameters. Learn more about it [here](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters).